.github/workflows/scripts/prepare-changelog.sh

```diff
@@ -50,17 +50,17 @@ convert_pr_issue_links() {
     local result="$content"
     
     # Find all (#number) patterns
-    local numbers=$(echo "$content" | grep -oE '\(#[0-9]+\)' | grep -oE '[0-9]+' | sort -u)
-    
+    local numbers=$(echo "$content" | grep -oE '\(#[0-9]+((, #[0-9]+)?)*\)' | grep -oE '[0-9]+' | sort -u)
+
     local total=$(echo "$numbers" | wc -w)
-    
+
     for number in $numbers; do
         local type=$(check_pr_or_issue "$repo_url" "$number")
-        
-        # Replace (#number) with proper markdown link
-        result=$(echo "$result" | sed "s|(#$number)|([#$number]($repo_url/$type/$number))|g")
+
+        # Replace #number with proper markdown link
+        result=$(echo "$result" | sed "s|#$number|[#$number]($repo_url/$type/$number)|g")
```

Comment:
`grep` is already filtering down possible matches to at least start the match with `(#number`, so I figure this isn't too dangerous of a regex modification on sed.

Also, for a given repository's changelog notes, the assumption is already that `#number` is a local repo's issue or PR, so even if there's a false positive (an issue or PR not surrounded by parentheses) it will still be the correct local repo reference.